core/test/presentation/views/text/rich_text_builder_test.dart

```diff
@@ -0,0 +1,208 @@
+import 'package:core/presentation/extensions/list_extensions.dart';
+import 'package:core/presentation/views/text/rich_text_builder.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  const normalTextStyle = TextStyle(
+    color: Colors.black,
+    fontSize: 16,
+  );
+  const highlightTextStyle = TextStyle(
+    color: Colors.red,
+    fontSize: 16,
+  );
+  Widget buildRichText({
+    required String textOrigin,
+    required String wordToStyle,
+    String? preMarkedText,
+    bool ensureHighlightVisible = false,
+  }) => MaterialApp(home: Scaffold(
+    body: RichTextBuilder(
+      textOrigin: textOrigin,
+      wordToStyle: wordToStyle,
+      styleOrigin: normalTextStyle,
+      styleWord: highlightTextStyle,
+      preMarkedText: preMarkedText,
+      ensureHighlightVisible: ensureHighlightVisible,
+    ),
+  ));
+
+  Finder findNormalStyledText(String text, {int occurence = 1}) {
+    final normalSpan = TextSpan(text: text, style: normalTextStyle);
+    return find.byWidgetPredicate((widget) {
+      return widget is RichText
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.contains(normalSpan)
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.countOccurences(normalSpan) == occurence;
+    });
+  }
+
+  Finder findHighlightStyledText(String text, {int occurence = 1}) {
+    final highlightSpan = TextSpan(text: text, style: highlightTextStyle);
+    return find.byWidgetPredicate((widget) {
+      return widget is RichText
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.contains(highlightSpan)
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.countOccurences(highlightSpan) == occurence;
+    });
+  }
+
+  group('rich text builder test:', () {
+    testWidgets(
+      'should not highlight any text '
+      'when wordToStyle is empty '
+      'and preMarkedText is null',
+    (tester) async {
+      // arrange
+      const textOrigin = 'This is a test';
+      final richText = buildRichText(
+        textOrigin: textOrigin,
+        wordToStyle: '',
+      );
+      
+      // act
+      await tester.pumpWidget(richText);
+      
+      // assert
+      expect(
+        findNormalStyledText(textOrigin),
+        findsOneWidget,
+      );
```

Comment:
should verify not find any highlight widgets

Reply:
Done